packages/code-block/partials/code-lines/index.tsx

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/**
+ * Note: lines of code are expected to be stable. If we need to work with
+ * dynamic code blocks in the future, we could assign random unique IDs
+ * to each line during the `linesOfCode` `useMemo` function.
+ *
+ * For now, we disable react/no-array-index key for the entire file.
+ */
+/* eslint-disable react/no-array-index-key */
+
+import React, { PropsWithChildren, ReactElement, useMemo } from 'react'
+import classNames from 'classnames'
+import parseHighlightedLines from '../../utils/parse-highlighted-lines'
+import splitJsxIntoLines from './utils/split-jsx-into-lines'
+import splitHtmlIntoLines from './utils/split-html-into-lines'
+import s from './code-lines.module.css'
+
+interface CodeLinesProps {
+  code: string | ReactElement
```

Comment:
Just curious: is there a specific reason you went with `ReactElement`? I _think_ `code: ReactNode` would be the more accurate type since that's inclusive of `string`.

Reply:
Great point! Intent was that we might want `string | ReactElement` as it gets a little more specific about `code` not being any of the other possible types under `ReactNode`:

```ts
type ReactNode = ReactElement | string | number | ReactFragment | ReactPortal | boolean | null | undefined;
```